docs/modules/configuration/pages/dynamic-config-persistence.adoc

```diff
@@ -0,0 +1,208 @@
+
+== Production Checklist
+
+Before you start configuring members to persist dynamic configuration changes, consider the following checklist:
+
+- [ ] Does your cluster have enough available disk space, especially if you want to <<backup, create backups of the configuration file>>?
+- [ ] Do you use <<snippets, configuration snippets>>? You can configure members to back up snippets instead of the <<glossary, root configuration file>>.
+- [ ] Does your cluster have read/write permissions for the configuration file? All members must have write permissions to the configuration file.
+- [ ] Is your cluster free of configuration conflicts? Hazelcast does not support persistence for dynamic configuration in clusters that have conflicting configuration files.
+
+NOTE: Running clusters must be restarted before any configuration changes take effect.
+
+
+[[backup]]
+== Creating Backups of the Configuration File
+
+By default, Hazelcast keeps 5 backups of the configuration file in the member's startup directory.
+
+To change the number of backups, configure the `backup-count` option.  To change the directory in which to store the backups, configure the <<dynamic-configuration-backup-dir,`backup-dir`>> option.
+
+The filenames of backups are in the following format, which includes the date and time when they were created followed by the `.backup` extension: `yyyy-MM-dd--HH-mm-ss-SSS.backup`. For example, `2022-01-31--13-23-44-699.backup`.
+
+WARNING: If the backup directory contains existing files with the `.backup` extension or if time is shifted while the member is running, the order of backups can be corrupted.
+
+If you configure the <<snippets, `persistence-file`>> option, that file is backed up instead of the <<glossary, root configuration file>>.
+
+[tabs]
+====
+XML::
++
+--
+[source,xml]
+----
+<hazelcast>
+  <dynamic-configuration>
+    <persistence-enabled>true</persistence-enabled>
+    <backup-dir>/data/dynamic/backups</backup-dir>
+    <backup-count>7</backup-count>
+  </dynamic-configuration>
+</hazelcast>
+----
+--
+YAML::
++
+--
+[source,yml]
+----
+hazelcast:
+  dynamic-configuration:
+    persistence-enabled: true
+    backup-dir: /data/dynamic/backups
+    backup-count: 7
+----
+--
+====
+
+[[snippets]]
+== Changing the Persistence File
+
+To persist dynamic configuration changes to a file other than the <<glossary, root configuration file>>, set the path to an XML or YAML configuration file in the `persistence-file` option.
+
+The `persistence-file` must be created and must be xref:configuring-declaratively.adoc#composing-declarative-configuration[imported] into the root configuration file.
+
+If you start multiple members on a single device, each member should have a unique persistence file.
+
+If you use the Java API to set the `Config.configFile` property, Hazelcast persists changes to that file.
```

Comment:
This isn't a property, this is a method. `Config.setConfigurationFile()`. Basically we want user to set `configurationFile` field of the `Config` object.